### PR TITLE
chore: update optimism

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

This PR adds instructions for running an op stack + celestia devnet locally, both with a v1.2.0 stable version and a version that is tracking upstream.

The `optimism.md` testnet guide will be turned into a "light node deployment" guide of some kind similar to optimism's deployment guide. This will be an example instead of something that is copy-pastable.

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x]  Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by tuxcanfly — APPROVED** [review]

File: developers/optimism-devnet.md [hunk 1]
```diff
@@ -22,157 +23,31 @@ git clone https://github.com/celestiaorg/optimism
...
+git checkout tux/rebase-v.1.2.0
 ```
 
-```bash-vue [Testnet]
-git checkout v0.1.3-OP_v1.0.6-CN_v0.11.0-rc8
+```bash-vue [rebase upstream]
+git checkout tux/rebase-upstream
+cd packages/contracts-bedrock/lib/forge-std/
+git submodule init && git submodule update
+cd ../../../../
```
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply developers/optimism-devnet.md:38]: https://github.com/ethereum-optimism/optimism/issues/8132

**Comment by jcstein** [comment]
one thing, want to make releases next week and use those?

File: developers/optimism-devnet.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tuxcanfly) [reply developers/optimism-devnet.md:38]: Fixed in upstream 🔥

[the remark this answers is not in the document]
Comment (tuxcanfly) [reply developers/optimism-devnet.md:38]: Edit: We can remove this hack now.

[the remark this answers is not in the document]
Comment (jcstein) [reply developers/optimism-devnet.md]: have changed to `git submodule update --init --recursive` until we have the commit from upstream in ours 🫡

**Pull request merged**